### ISSUE TITLE
Use Slog logger

### ISSUE THREAD

**Issue by mehdihadeli** [mehdihadeli/store-golang-microservices#65]
- [https://github.com/golang/go/issues/56345](https://github.com/golang/go/issues/56345)
- [https://github.com/lmittmann/tint](https://github.com/lmittmann/tint)

**Comment by Alan-MQ** [comment]
I wanna try this one too, but I'm not sure if I am running this project correctly~~,
just run cmd/app/main.go from catalog_read_service, write_service and order_service?

**Comment by Alan-MQ** [comment]
And I notice we will intergate ELK to this project,  so we might wanna Use Slog before we introduce ELK right?

**Comment by Alan-MQ** [comment]
> And I notice we will intergate ELK to this project, so we might wanna Use Slog before we introduce ELK right? and by the way, intergate ELK need someone fully understand this project right? about how fx are working here and event driven , message queue all that stuff~~~

try to run this project with this method and I got something like this 
![image](https://github.com/mehdihadeli/go-ecommerce-microservices/assets/109934126/c3d4e19e-b7a7-4460-9f2b-1975bbc21c01)

but , when I `make unit-test` locally all tests passed. that means the docker enviroment was nothing wrong right?

**Comment by mehdihadeli** [comment]
Yes, slog is built structured logging in go 21.0, so it is better we switch to this library. Maybe we have to upgrade the project to go 21.0 at first. Maybe in a separated issue or pull request. 
Yes, first we should use slog after that we integrate that with ELK and kibana

**Comment by mehdihadeli** [comment]
> > And I notice we will intergate ELK to this project, so we might wanna Use Slog before we introduce ELK right? and by the way, intergate ELK need someone fully understand this project right? about how fx are working here and event driven , message queue all that stuff~~~
> 
> try to run this project with this method and I got something like this ![image](https://user-images.githubusercontent.com/109934126/264572503-c3d4e19e-b7a7-4460-9f2b-1975bbc21c01.png)
> 
> but , when I `make unit-test` locally all tests passed. that means the docker enviroment was nothing wrong right?

This is a temp docker container that use by go-testcotainer library for my integration and end-to-end tests, it should be ephemeral and after running tests it should be destroyed. `65027:27017` is a testcontainer for MongoDB

**Comment by mehdihadeli** [comment]
Could you first upgrade to version 21 at first #81 ?

**Comment by Alan-MQ** [comment]
ok, so I will look up all this parpers you mentioned on the top , and come up with a plan about how to use slog, but at some point , I'm gonna need to run this project correctly to see the log right ? is there any doc on how should I run this project?

**Comment by Alan-MQ** [comment]
> 

ok， I will check it out first.
